### PR TITLE
pkg: relax version constraints for ocamlformat dev tool

### PR DESCRIPTION
**Pull request by gridbugs**
Previously dune would install the exact version of ocamlformat specified in the .ocamlformat file in the project's root. This prevents alternative distributions of the ocamlformat package from being used, as these typically append a suffix to the version number.

The motivation for this change is supporting a binary dev tools distribution where the ocamlformat package is named like "ocamlformat.0.26.2+binary", and we want that package to be installable when the .ocamlformat file of a project specifies "version=0.26.2".

### CODE REVIEW

File: bin/lock_dev_tool.ml [hunk 1]
```diff
@@ -13,6 +13,31 @@ let is_enabled =
...
+  (* The goal here is to add a suffix to [min_version] to construct a version
+     number higher than than any version number likely to appear with
+     [min_version] as a prefix. "_" is the highest ascii symbol that can appear
+     in version numbers, excluding "~" which has a special meaning. It's
+     conceivable that one or two consecutive "_" characters may be used in a
+     version, so this appends "___" to [min_version].
+
+     Read more at: https://opam.ocaml.org/doc/Manual.html#Version-ordering
+  *)
+  let max_version = String.concat ~sep:"" [ min_version; "___" ] in
```
Comment (Leonidas-from-XIV) [inline bin/lock_dev_tool.ml:32]:
```suggestion
  let max_version = min_version ^ "___"  in
```

**Review by Leonidas-from-XIV — COMMENTED** [review]
Seems sensible to me, I would assume `ocamlformat.0.26.2.1`should qualify if one requests `0.26.2`.

**Comment by moyodiallo** [comment]
Would you like add a test ?

**Comment by gridbugs** [comment]
Added a test

**Review by Leonidas-from-XIV — APPROVED** [review]
LGTM! Nice and clear test.

**Pull request merged**